### PR TITLE
Frontend/compact state config

### LINKED ISSUE #782: state self-serve config FE

### PR DESCRIPTION
**Pull request by jsandoval81**
### Requirements List
- _None_

### Description List
- Added component InputEmailList
- Added component CompactSettingsConfig
- Added component CompactStateConfig
- Updated Compact settings page with CompactSettingsConfig component + conditional state list
    - If user has compact-admin, they will see the existing payment-gateway credentials form + the new compact config form
    - If user has compact-admin and any state-admin, they will also see a list of states they can navigate to edit
    - If user has no compact-admin and multiple state-admin, they will see only the list of states they can navigate to edit
    - If user has no-compact-admin and exactly 1 state-admin, they will be auto-redirected to the state config page
    - If user has no compact-admin and no state-admin, they will be auto-redirected to their home page
    - Updated existing payment-gateway form to make submit button full-width to match updated designs
- Added StateSettings page with CompactStateConfig component
    - If user does not have state-admin for the state, they will be auto-redirected to their home page
- Added label subtext FormInput model & InputText component
- Added disabled state & label subtext + changed primary color of InputRadioGroup to match updated designs
- Updated size of toggle icon to be smaller in InputPassword component 
- Added a warning state to the InputButton component
- Updated form validation handling for numbers & currency
- Updated main navigation to show Settings link if compact or state admin
- Updated Compact model with new interfaces for compact-config & state-config
- Updated the data API and mock data modules with new compact & state config calls

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- Code review
- Testing
    - Compact Settings page
        - Update mock data or local sand box to test the following scenarios:
            - If user has compact-admin, they will see the existing payment-gateway credentials form + the new compact config form
            - If user has compact-admin and any state-admin, they will also see a list of states they can navigate to edit
            - If user has no compact-admin and multiple state-admin, they will see only the list of states they can navigate to edit
            - If user has no-compact-admin and exactly 1 state-admin, they will be auto-redirected to the state config page
            - If user has no compact-admin and no state-admin, they will be auto-redirected to their home page
       - Compact settings form
           - Should match designs
           - Should show confirmation modal if changing compact from not-live to live
           - Should not allow making compact not-live after it has been set live (input should be disabled)
       - State list
           - If available, should take user to setting page for that state
    - State settings page
        - If user does not have state-admin for the state, they will be auto-redirected to their home page
        - Should match designs
        - Should show confirmation modal if changing compact from not-live to live
        - Should not allow making compact not-live after it has been set live (input should be disabled)

Closes #782 

## Summary by CodeRabbit

- **New Features**
  - Introduced Compact and State Settings management with new pages and components for configuring compact-level and state-specific settings.
  - Added interactive forms for fees, notification emails, privilege purchase toggles, and live status with confirmation dialogs and keyboard accessibility.
  - Enabled role-based navigation and access for compact and state admins, including a state list for users with multiple state admin roles.
  - Added email list input component and enhanced input buttons with warning styles.

- **Improvements**
  - Enhanced input components with label subtext, improved currency formatting, and better disabled state visuals.
  - Expanded English and Spanish localization with new validation messages and detailed settings UI text.
  - Refined permission checks and navigation to restrict access based on user roles.

- **Bug Fixes**
  - Improved form accessibility, focus trapping in modals, and consistent error messaging.

- **Tests**
  - Added unit and component tests for new components, models, and form mixins.

- **Style**
  - Added and updated stylesheets for new components and pages, ensuring responsive design and UI consistency.

### CODE REVIEW

File: webroot/src/components/Forms/InputEmailList/InputEmailList.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import InputEmailList from '@components/Forms/InputEmailList/InputEmailList.vue';
+
+describe('InputEmailList component', async () => {
+    it('should mount the component', async () => {
+        const wrapper = await mountShallow(InputEmailList);
+
+        expect(wrapper.exists()).to.equal(true);
+        expect(wrapper.findComponent(InputEmailList).exists()).to.equal(true);
+    });
+});
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/Forms/InputEmailList/InputEmailList.spec.ts:19]: Mount testing is the general component testing strategy for this app. It meets our test coverage needs and allows us to focus more on testing the data layer, such as the models and store. There are no plans to add additional tests to this component in this PR.

File: webroot/src/components/StateSettingsConfig/StateSettingsConfig.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import StateSettingsConfig from '@components/StateSettingsConfig/StateSettingsConfig.vue';
+
+describe('StateSettingsConfig component', async () => {
+    it('should mount the component', async () => {
+        const wrapper = await mountShallow(StateSettingsConfig);
+
+        expect(wrapper.exists()).to.equal(true);
+        expect(wrapper.findComponent(StateSettingsConfig).exists()).to.equal(true);
+    });
+});
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/StateSettingsConfig/StateSettingsConfig.spec.ts:19]: Mount testing is the general component testing strategy for this app. It meets our test coverage needs and allows us to focus more on testing the data layer, such as the models and store. There are no plans to add additional tests to this component in this PR.

File: webroot/src/components/Forms/InputEmailList/InputEmailList.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        if (Array.isArray(formInput.value)) {
+            (formInput.value as Array<string>) = formInput.value.filter((email) => email !== emailToRemove);
+        }
+
+        formInput.isTouched = true;
+        formInput.validate();
+        emailInput.focus();
+        formInput.isTouched = false;
+    }
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/Forms/InputEmailList/InputEmailList.ts:89]: The existing code compiles without error, while the the Typescript compiler throws an error if the suggested update is made. I agree the code would be better without having to typecast here, but this seems to be one of those cases where Typescript is having trouble with types on nested properties; and the existing code seems to be the least verbose way around that. I'm still open to suggestions if there are solid alternatives, otherwise this casting may have to be left in place.

File: webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+            if (this.formData[licenseTypeMilitary].value) {
+                this.formatBlur(this.formData[licenseTypeMilitary]);
+            }
+
+            this.feeInputs.push(this.formData[licenseType]);
+            this.feeInputs.push(this.formData[licenseTypeMilitary]);
+        });
+
+        this.watchFormInputs(); // Important if you want automated form validation
+    }
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts]: Just split the privilege fee inputs into their own method since they were their own large chunk of code. I left the core static inputs as-is and consistent with other form components.

File: webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+
+        stateConfig?.privilegeFees?.forEach((privilegeFee) => {
+            const licenseType = licenseTypes.find((type) => type.abbrev === privilegeFee.licenseTypeAbbreviation);
+
+            privilegeFee.name = licenseType?.name || privilegeFee.licenseTypeAbbreviation?.toUpperCase() || '';
+        });
+        this.initialStateConfig = stateConfig;
+        this.isPurchaseEnabledInitialValue = this.initialStateConfig?.licenseeRegistrationEnabled;
+        this.isLoading = false;
+    }
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts]: Just added a check for an error message and handle the appropriate lines conditionally.

File: webroot/src/pages/CompactSettings/CompactSettings.ts [hunk 6]
```diff
@@ -5,16 +5,136 @@
...
+                state: abbrev,
+            },
+        };
+
+        if (isRouteReplace) {
+            this.$router.replace(routeConfig);
+        } else {
+            this.$router.push(routeConfig);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/pages/CompactSettings/CompactSettings.ts]: Just added a conditional check to only perform the router action if the compact value is known.

**Comment by ChiefStief** [comment]
For the military fees we replace the value with a zero but ultimately send up null if this value wasn't filled in. When we refresh the page, its empty again. Ideally, what you see after sending the form matches what the form looks like when you refresh the page imo.

 What does being empty signify?
1) The military fee is zero ie military affiliates get compacts for free
2) The military fee is identical to the regular fee

Depending on the answer there I would suggest different fixes to the behavior.

**Comment by jsandoval81** [comment]
@ChiefStief 
> What does being empty signify?
>
> 1. The military fee is zero ie military affiliates get compacts for free
> 2. The military fee is identical to the regular fee

If the military fee is never meant to be 0 (free), then the existing implementation may work. If OTTH the military fee _can_ be free then there might be some additional work to un-conflate the number+null type here. We can check with the server devs offline about this.

_Side note: this is why dealing with types that are number+null is tricky - and we avoid where possible_ 🙂

**Review by ChiefStief — APPROVED** [review]

**Comment by jsandoval81** [comment]
@jlkravitz This is ready for your review.

File: webroot/src/network/stateApi/data.api.ts [hunk 7]
```diff
@@ -116,6 +116,46 @@ export class StateDataApi implements DataApiInterface {
...
+    /**
+     * PUT State config.
+     * @param  {string}             compact The compact string ID (aslp, octp, coun).
+     * @param  {string}             state   The 2-character state abbreviation.
+     * @param  {CompactStateConfig} config  The compact config data.
+     * @return {Promise<object>}            The server response.
+     */
+    public updateCompactStateConfig(compact: string, state: string, config: CompactStateConfig) {
+        return this.api.put(`v1/compacts/${compact}/jurisdictions/${state}`, config);
+    }
```
[the remark this answers is not in the document]
Comment (jlkravitz) [reply webroot/src/network/stateApi/data.api.ts]: @jsandoval81 Reading through this and seeing that coderabbit misspelled COUNSILING, I'm realizing we misspelled it in `webroot/src/models/Compact/Compact.model.ts`. Can we update that?

File: webroot/src/pages/StateSettings/StateSettings.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    //
+    // Watch
+    //
+    @Watch('currentCompact') currentCompactUpdate() {
+        this.permissionRedirectCheck();
+    }
+
+    @Watch('user') userUpdate() {
+        this.permissionRedirectCheck();
+    }
```
Comment (jlkravitz) [inline webroot/src/pages/StateSettings/StateSettings.ts:119]: What circumstances require a watch on the user object?

File: webroot/src/pages/CompactSettings/CompactSettings.ts [hunk 9]
```diff
@@ -5,16 +5,138 @@
...
+    //
+    // Methods
+    //
+    init(): void {
+        this.permissionRedirectCheck();
+    }
+
+    permissionRedirectCheck(): void {
+        if (this.currentCompact && this.user) {
+            if (!this.isLoggedInAsStaff || (!this.isCompactAdmin && !this.isStateAdminAny)) {
```
Comment (jlkravitz) [inline webroot/src/pages/CompactSettings/CompactSettings.ts:104]: nit: I believe this could also work since isLoggedInAsStaff is embedded in the computation of the other properties.
 
```suggestion
            if (!this.isCompactAdmin && !this.isStateAdminAny) {
```

Comment (jlkravitz) [reply webroot/src/pages/CompactSettings/CompactSettings.ts:104]: From there, I think you could hoist the `isCompactAdmin` check one level up to get something like the below pseudo-code.

```python
if currentCompact & user & !isCompactAdmin
    if !isStateAdminAny

    else if isStateAdminExactlyOne

    end
end
```

File: webroot/src/pages/StateSettings/StateSettings.ts [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    //
+    // Methods
+    //
+    init(): void {
+        this.permissionRedirectCheck();
+    }
+
+    permissionRedirectCheck(): void {
+        if (this.currentCompact && this.user) {
+            if (!this.isLoggedInAsStaff || !this.isStateAdmin) {
```
Comment (jlkravitz) [inline webroot/src/pages/StateSettings/StateSettings.ts:99]: nit: I believe this could also work since isLoggedInAsStaff is embedded in the computation of `isStateAdmin` (though open to arguments that prefer the explicitness!)

```suggestion
            if (!this.isStateAdmin) {
```

File: webroot/src/models/_formatters/currency.ts [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+ * @return {string}
+ */
+const formatCurrencyInput = (value: string | number = ''): string => {
+    let [ dollars, cents ] = value.toString().split(/\.(.*)/s);
+    const hasDecimal = cents !== undefined;
+    let formatted = '';
+
+    // Get raw dollar & cent values
+    dollars = dollars.replace(/\D/g, '');
+    cents = (hasDecimal) ? cents.replace(/\D/g, '') : '';
```
Comment (jlkravitz) [inline webroot/src/models/_formatters/currency.ts:20]:
```suggestion
    cents = hasDecimal ? cents.replace(/\D/g, '') : '';
```

File: webroot/src/models/_formatters/currency.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    } else if (cents) {
+        formatted = `0.${cents}`;
+    }
+
+    return formatted;
+};
+
+/**
+ * Currency formatter for completed input.
+ * @param  {string | number} value              The active input value.
```
Comment (jlkravitz) [inline webroot/src/models/_formatters/currency.ts:41]: right?

```suggestion
 * @param  {string | number} value              The value held by the blurred input.
```

File: webroot/src/models/_formatters/formatters.spec.ts [hunk 13]
```diff
@@ -84,3 +85,65 @@ describe('Phone formatters', () => {
...
+
+        expect(formatted).to.equal('0.11');
+    });
+    it('should return empty for blur when value param is empty (blur)', () => {
+        const formatted = formatCurrencyBlur();
+
+        expect(formatted).to.equal('');
+    });
+    it('should return return correctly formatted when value param is empty and optional (blur)', () => {
+        const formatted = formatCurrencyBlur('', true);
```
Comment (jlkravitz) [inline webroot/src/models/_formatters/formatters.spec.ts:125]: Can we add a test for what should happen when param is _not_ optional and value is empty?

File: webroot/src/models/_formatters/formatters.spec.ts [hunk 14]
```diff
@@ -84,3 +85,65 @@ describe('Phone formatters', () => {
...
+    });
+    it('should return correctly formatted when value param is only dollars (blur)', () => {
+        const formatted = formatCurrencyBlur('1');
+
+        expect(formatted).to.equal('1');
+    });
+    it('should return correctly formatted when value param is only dollars with trailing decimal point (blur)', () => {
+        const formatted = formatCurrencyBlur('1.');
+
+        expect(formatted).to.equal('1');
```
Comment (jlkravitz) [inline webroot/src/models/_formatters/formatters.spec.ts:142]: can you document why the blur function removes the trailing decimal point, but the input formatter does not? (perhaps makes most sense to document in `currency.ts` itself in the function comments.

File: webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts [hunk 15]
```diff
@@ -0,0 +1,399 @@
...
+        };
+
+        // Call the server API to update
+        await dataApi.updateCompactStateConfig(compact, this.stateAbbrev, payload).catch((err) => {
+            this.setError(err.message);
+        });
+
+        // Handle success
+        if (!this.isFormError) {
+            if (Object.prototype.hasOwnProperty.call(payload, 'licenseeRegistrationEnabled')) {
```
Comment (jlkravitz) [inline webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts:324]: Can we add a brief comment re: why this fancy prototype code is necessary?

**Review by jlkravitz — COMMENTED** [review]
looks good! a few questions

File: webroot/src/network/stateApi/data.api.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/network/stateApi/data.api.ts]: @jlkravitz Good catch. Fixed.

File: webroot/src/pages/StateSettings/StateSettings.ts (hunk 8 above)
Comment (jsandoval81) [reply webroot/src/pages/StateSettings/StateSettings.ts:117]: Any case where the remote fetch of the user data from the server is still waiting while the rest of the component has loaded. A common case of this is a page refresh.

File: webroot/src/pages/CompactSettings/CompactSettings.ts (hunk 9 above)
Comment (jsandoval81) [reply webroot/src/pages/CompactSettings/CompactSettings.ts]: Updated!

File: webroot/src/pages/StateSettings/StateSettings.ts (hunk 10 above)
Comment (jsandoval81) [reply webroot/src/pages/StateSettings/StateSettings.ts]: Updated

File: webroot/src/models/_formatters/currency.ts (hunk 11 above)
Comment (jsandoval81) [reply webroot/src/models/_formatters/currency.ts:21]: Our code style has been to include parens around the conditional in ternary statements. We find it to be easier to read, but don't feel strongly if you would prefer this case to remove the parens.

File: webroot/src/models/_formatters/formatters.spec.ts (hunk 14 above)
Comment (jsandoval81) [reply webroot/src/models/_formatters/formatters.spec.ts:142]: Added JSDoc comments to those functions, more about their general separation than the specific trailing decimal case. Let me know if you'd like to see additional comments.

File: webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts [hunk 16]
```diff
@@ -0,0 +1,399 @@
...
+            privilegeFees: feeInputsCore.map((feeInputCore) => {
+                // Map indeterminate set of privilege fee inputs to their payload structure
+                const [ licenseType ] = feeInputCore.id.split('-');
+                const militaryInput = Object.values(this.formData).find((formInput) =>
+                    (formInput as unknown as FormInput).id === `${feeInputCore.id}-military`) as FormInput | undefined;
+
+                return {
+                    licenseTypeAbbreviation: licenseType,
+                    amount: Number(feeInputCore.value),
+                    militaryRate: ([null, ''].includes(militaryInput?.value as any)) ? null : Number(militaryInput?.value),
```
[the remark this answers is not in the document]
Comment (jsandoval81) [reply webroot/src/components/StateSettingsConfig/StateSettingsConfig.ts]: Updated

File: webroot/src/models/_formatters/currency.ts (hunk 11 above)
Comment (jlkravitz) [reply webroot/src/models/_formatters/currency.ts:21]: Ah, didn't realize it was our standard. I have a slight preference for no parens in cases like these, but there's no rhyme or reason to that preference. Consistency is more important so let's keep as-is!

File: webroot/src/models/_formatters/formatters.spec.ts (hunk 13 above)
Comment (jsandoval81) [reply webroot/src/models/_formatters/formatters.spec.ts:130]: I just added an extra test with that case (option 1).

**Comment by jsandoval81** [comment]
@jlkravitz This is ready for your re-review.

**Review by jlkravitz — APPROVED** [review]
@isabeleliassen good to merge

**Review by isabeleliassen — APPROVED** [review]
